app/views/shared/_block_link.html.erb

```diff
@@ -2,10 +2,34 @@
 yields: Link text (required).
 locals:
 * url: URL of link.
+* external: Whether link is external. Defaults to true if hostname is different from request.
 %>
-<% content = yield.presence or raise "no block content given" %>
-<%= tag.a href: url, class: 'usa-link block-link' do %>
+<%
+content = yield.presence or raise "no block content given"
+external = local_assigns.fetch(:external) do
+  url_host_with_port = URI.parse(url).select(:host, :port).compact.join(':')
+  url_host_with_port.present? && url_host_with_port != "#{request.host}:#{request.port}"
+end
+tag_attrs = { href: url, class: 'usa-link block-link' }
+tag_attrs[:target] = '_blank' if external
```

Comment:
I thought our whole thing for external links was that we prefix them with text but we still open them in the same window?

Reply:
The argument being made in the accessibility discussion and [related Slack discussion](https://gsa-tts.slack.com/archives/CNCGEHG1G/p1629130064005400?thread_ts=1629129400.004500&cid=CNCGEHG1G) is that navigating to an external link in the IAL2 flow satisfies some exceptions to [the rule](https://www.w3.org/TR/WCAG20-TECHS/G200.html), notably "significantly disrupt a multi-step workflow, such as filling in and submitting a form, if the page is opened in the same window or tab" and "provides extensive context-sensitive help [...] in new windows or tabs to prevent the loss of any form data that has already been entered."

Though as I write it out, and considering that the usage of a partial like this shouldn't be exclusive to IAL2, maybe we should at least drop the implicit "new tab if different host" behavior and pass the option explicitly everywhere it's relevant within the IAL2 flow.

Reply:
> Though as I write it out, and considering that the usage of a partial like this shouldn't be exclusive to IAL2, maybe we should at least drop the implicit "new tab if different host" behavior and pass the option explicitly everywhere it's relevant within the IAL2 flow.

Updated in bd5a8e2, with additional clarification around "new tab" not being strictly relevant to the links being external, and should be used within guidelines mentioned above.

Reply:
Makes sense to me! Thx 👍